### PR TITLE
enhance: Support dynamic config for opentelemetry trace

### PR DESCRIPTION
**Pull request by aoiasd**
relate: https://github.com/milvus-io/milvus/issues/31940

### CODE REVIEW

File: cmd/roles/roles.go [hunk 1]
```diff
@@ -392,6 +394,30 @@ func (mr *MilvusRoles) Run() {
...
+			log.Warn("Close old provider failed, stop reset", zap.Error(err))
+			return
+		}
+
+		tracer.SetTracerProvider(exp, params.TraceCfg.SampleFraction.GetAsFloat())
+		log.Info("Reset tracer finished", zap.String("Exporter", params.TraceCfg.Exporter.GetValue()))
+
+		if paramtable.GetRole() == typeutil.QueryNodeRole || paramtable.GetRole() == typeutil.StandaloneRole {
+			initcore.InitTraceConfig(params)
+			log.Info("Reset segcore tracer finished", zap.String("Exporter", params.TraceCfg.Exporter.GetValue()))
```
Comment (congqixia) [inline cmd/roles/roles.go:418]: cpp tracer is not updated, since cpp use `call_once` inside

**Comment by aoiasd** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**